### PR TITLE
Add git --signoff to contributing.md

### PR DESCRIPTION
**Pull request by AriESQ**
Update contributing.md to require `git -signoff`

This repo will reject commits that do not have the `git -signoff` information in them, via the DCO github action. This PR updated contributing.md to let contributors know they must include a signoff.

# License Assignment

This repository is copyright 2017-2023 Joseph Block under a [Attribution-NonCommercial-ShareAlike 4.0 International](#attribution-noncommercial-sharealike-40-international) license.

- [x] I assign copyright to the changes in this PR to Joseph Block per the [Attribution-NonCommercial-ShareAlike 4.0 International](#attribution-noncommercial-sharealike-40-international) license.

# Types of changes

- [ ] New entry
- [ ] Typo cleanup
- [x] Text updates

# Checklist:

- [x] All new and existing tests pass.
- [x] List entries are sorted alphabetically.
- [x] Any list entries are one line. This makes it easier to keep the sections sorted. Let the markdown renderer handle line breaks, don't try to help it.
- [x] I confirmed that all the links in my PR are valid.

### CODE REVIEW

**Comment by AriESQ** [comment]
If someone doesn't know to do a `git -signoff` commit; the DCO github action will give them the following error; which requires that they use a local git environment to rebase the branch. This is kind of sub-optimal, given that folks may want to just do a drive-by edit with the Github GUI editor. A benefit of using the Github GUI editor is that it verifies the committer, and populates the anti-spam email relay automatically.  

I might reconsider whether you want to require this. The PR template has a sign-off/assignment, which should be sufficient. If a friendly committer messes up the `--signoff` they may abandon the PR instead of going through the trouble of a whole rebase and resubmit. 

> DCO / DCO required action Oct 25, 2023 in 1s
> DCO
> 
> There is one commit incorrectly signed off. This means that the author of this commit failed to include a Signed-off-by line in the commit message.
> 
> To avoid having PRs blocked in the future, always include Signed-off-by: Author Name <authoremail@example.com> in every commit message. You can also do this automatically by using the -s flag (i.e., git commit -s).
> 
> Here is how to fix the problem so that this code can be merged.
> Rebase the branch
> 
> If you have a local git environment and meet the criteria below, one option is to rebase the branch and add your Signed-off-by lines in the new commits. Please note that if others have already begun work based upon the commits in this branch, this solution will rewrite history and may cause serious issues for collaborators ([described in the git documentation](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) under "The Perils of Rebasing").
> 
> You should only do this if:
> 
>     You are the only author of the commits in this branch
>     You are absolutely certain nobody else is doing any work based upon this branch
>     There are no empty commits in the branch (for example, a DCO Remediation Commit which was added using --allow-empty)
> 
> To add your Signed-off-by line to every commit in this branch:
> 
>     Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
>     In your local branch, run: git rebase HEAD~1 --signoff
>     Force push your changes to overwrite the branch: git push --force-with-lease origin yubikey2
> 
> Summary
> 
> Commit sha: [2df77e9](https://github.com/unixorn/sysadmin-reading-list/pull/155/commits/2df77e91dc8c94a7b106a2644d5e10e0eeb32184), Author: AriESQ, Committer: GitHub; The sign-off is missing.

**Review by unixorn — APPROVED** [review]
lgtm

**Comment by unixorn** [comment]
Thanks!

**Pull request merged**
